kernel/fs/inode.rs

```diff
@@ -156,6 +156,10 @@ pub trait FileLike: Send + Sync + Downcastable {
     ) -> Result<(usize, SockAddr)> {
         Err(Error::new(Errno::EBADF))
     }
+
+    fn chmod(&self, _mode: FileMode) -> Result<()> {
+        Err(Error::new(Errno::EBADF))
```

Comment:
I'd use `ENOSYS` because [`EBADF` is used for different reasons in Linux](https://man7.org/linux/man-pages/man2/chmod.2.html):
```suggestion
        Err(Error::new(Errno::ENOSYS))
```